service/src/main/kotlin/fi/espoo/evaka/invoicing/controller/FinanceBasicsController.kt

```diff
@@ -158,6 +160,50 @@ class FinanceBasicsController(
             .also { Audit.FinanceBasicsVoucherValuesRead.log() }
     }
 
+    @PostMapping("/voucher-values")
+    fun createVoucherValue(
+        db: Database,
+        user: AuthenticatedUser,
+        clock: EvakaClock,
+        @RequestBody body: ServiceNeedOptionVoucherValueRange
+    ) {
+        val id =
+            db.connect { dbc ->
+                dbc.transaction { tx ->
+                    accessControl.requirePermissionFor(
+                        tx,
+                        user,
+                        clock,
+                        Action.Global.CREATE_VOUCHER_VALUE
+                    )
+
+                    val serviceNeedOption =
+                        tx.getServiceNeedOptions()
+                            .filter { it.id == body.serviceNeedOptionId }
+                            .firstOrNull()
```

Comment:
nitpick: could be shortened to `tx.getServiceNeedOptions().find { it.id == body.serviceNeedOptionId }`